drivers: ethernet: enc424j600: Fix interrupt handling

nordicjm opened the pull request:
This fixes an issue with interrupts being triggered before the
enc424j600 driver is ready to receive them or has configured the
interrupt sources which would prevent the driver from loading. An
optional Kconfig option has been added to allow not re-enabling
interrupts after an unknown interrupt has occured, this brings the
driver in-line with all other Zephyr ethernet drivers and the Linux
kernel encx24j600 driver.

Signed-off-by: Jamie McCrae <jamie.mccrae@lairdconnect.com>

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/35091

Output after change using dumb_http_server sample app:
```
*** Booting Zephyr OS build zephyr-v2.4.0-3426-g70ec734b87c3  ***
[00:00:00.646,453] <dbg> ethdrv.enc424j600_init_filters: ERXFCON: 0x005b
[00:00:00.646,789] <dbg> ethdrv.enc424j600_init_phy: PHANA: 0x05e1
[00:00:00.647,003] <dbg> ethdrv.enc424j600_init_phy: PHCON1: 0x1200
[00:00:00.647,216] <dbg> ethdrv.enc424j600_init: EIE: 0x8850
[00:00:00.647,277] <dbg> ethdrv.enc424j600_init: ECON1: 0x0001
[00:00:00.647,277] <err> ethdrv: handling int
[00:00:00.647,399] <inf> ethdrv: ENC424J600 Initialized
[00:00:00.647,430] <dbg> ethdrv.enc424j600_rx_thread: ESTAT: 0xda00
[00:00:00.647,491] <inf> ethdrv: Link down
[00:00:00.647,521] <err> ethdrv: handling int
[00:00:00.647,613] <dbg> ethdrv.enc424j600_rx_thread: ESTAT: 0x5a00
[00:00:00.647,644] <err> ethdrv: Unknown Interrupt, EIR: 0x0700
[00:00:00.653,533] <inf> net_config: Initializing network
[00:00:00.653,533] <inf> net_config: Waiting interface 1 (0x20000f0c) to be up...
Single-threaded dumb HTTP server waits for a connection on port 8080...
[00:00:02.244,201] <err> ethdrv: handling int
[00:00:02.244,323] <dbg> ethdrv.enc424j600_rx_thread: ESTAT: 0xdf00
[00:00:02.244,354] <inf> ethdrv: Link up
[00:00:02.244,537] <dbg> ethdrv.enc424j600_setup_mac: PHANLPA: 0x85e1
[00:00:02.244,750] <inf> ethdrv: 100Mbps
[00:00:02.244,750] <inf> ethdrv: full duplex
[00:00:02.244,873] <dbg> ethdrv.enc424j600_setup_mac: MACON2: 0x40b3
[00:00:02.244,934] <dbg> ethdrv.enc424j600_setup_mac: MAMXFL (maximum frame length): 1518
[00:00:02.244,964] <inf> ethdrv: Not suspended
[00:00:02.244,995] <inf> net_config: Interface 1 (0x20000f0c) coming up
[00:00:02.245,086] <inf> net_config: IPv4 address: 192.168.1.55
```

lairdjm commented:
Output from above was using nRF connect SDK, output below is from using latest master mainline zephyr branch (cable was connected at startup, then removed, then put back):
```
*** Booting Zephyr OS build v2.6.0-rc1-34-g977aba6d02c8  ***
[00:00:00.552,703] <inf> ethdrv: Link down
[00:00:00.552,764] <inf> ethdrv: ENC424J600 Initialized
[00:00:00.558,227] <inf> net_config: Initializing network
[00:00:00.558,258] <inf> net_config: Waiting interface 1 (0x200008d0) to be up...
Single-threaded dumb HTTP server waits for a connection on port 8080...
[00:00:02.076,995] <inf> ethdrv: Link up
[00:00:02.077,209] <inf> ethdrv: 100Mbps
[00:00:02.077,209] <inf> ethdrv: full duplex
[00:00:02.077,362] <inf> ethdrv: Not suspended
[00:00:02.077,392] <inf> net_config: Interface 1 (0x200008d0) coming up
[00:00:02.077,453] <inf> net_config: IPv4 address: 192.0.2.1
[00:00:08.695,404] <inf> ethdrv: Link down
[00:00:11.210,052] <inf> ethdrv: Link up
[00:00:11.210,235] <inf> ethdrv: 100Mbps
[00:00:11.210,235] <inf> ethdrv: full duplex
[00:00:11.210,388] <inf> ethdrv: Not suspended
```

jfischer-no commented on drivers/ethernet/eth_enc424j600.c:
```diff
@@ -484,7 +484,6 @@ static void enc424j600_rx_thread(struct enc424j600_runtime *context)
 			}
 		} else {
 			LOG_ERR("Unknown Interrupt, EIR: 0x%04x", eir);
-			continue;
```
How this fixes a bug?

pfalcon replied:
This explanation makes sense to me, but it should be in the commit message. (Description of how it behaved incorrectly before and how it'll behave correctly after this change.)

Commit:
drivers: ethernet: enc424j600: Fix interrupt handling

This fixes an issue with interrupts being triggered before the
enc424j600 driver is ready to receive them or has configured the
interrupt sources which would prevent the driver from loading. An
optional Kconfig option has been added to allow not re-enabling
interrupts after an unknown interrupt has occured, this brings the
driver in-line with all other Zephyr ethernet drivers and the Linux
kernel encx24j600 driver.

Signed-off-by: Jamie McCrae <jamie.mccrae@lairdconnect.com>

jfischer-no commented on drivers/ethernet/eth_enc424j600.c:
```diff
@@ -484,7 +485,6 @@ static void enc424j600_rx_thread(struct enc424j600_runtime *context)
 			}
 		} else {
 			LOG_ERR("Unknown Interrupt, EIR: 0x%04x", eir);
-			continue;
```
NACK, this is intended as explained [here]( https://github.com/zephyrproject-rtos/zephyr/issues/35091#issuecomment-839813047)

lairdjm replied:
This should be behind a Kconfig option that is disabled by default. A driver should not screw itself up because it received an event that it can safely ignore. Query: should an AT modem completely shut down if it receives a wrong command or emit an error and continue working as normal? This is useful for driver development and debugging but is nothing but extremely unhelpful in a production-grade firmware for an IoT device

jfischer-no replied:
No, it must not be ignored. To be clear, alternative to continue is k_oops()

lairdjm replied:
This is not in-line with any other ethernet driver in zephyr nor the encx24j600 driver in the linux kernel, it is staying to keep all these drivers operating in the same expected way.

jfischer-no replied:
It is nonsense. Do not change it, do not introduce new config option. And please do not resolve unresolved conversation.

jfischer-no commented on drivers/ethernet/eth_enc424j600.c:
```diff
@@ -450,13 +450,14 @@ static void enc424j600_rx_thread(struct enc424j600_runtime *context)
 				     ENC424J600_SFRX_ESTATL, &estat);
 		LOG_DBG("ESTAT: 0x%04x", estat);
 
+		enc424j600_read_sfru(context->dev, ENC424J600_SFRX_EIRL, &eir);
```
NACK, this is wrong, do not change it, first INTIE must be disabled.

jfischer-no reviewed (changes requested):
This patch does not fix  anything and is only a workaround for an unidentified problem.

lairdjm commented:
I have an old olimax enc624j600 board which I attached to our BL5340 development board to use instead of the on-board enc424j600 to exclude any possible issues with the components on our board, this made no difference and I get the unknown interrupt error whilst at the same time someone else remove all the intermediatory components on our development board between the BL5340 and enc424j600 e.g. for level shifting which also made no difference. I then tested with an nRF5340-DK and the olimax enc624j600 and I see the exact same issue with the original code - it does not work:
```
\1b[1;32muart:~$ \1b[m*** Booting Zephyr OS build zephyr-v2.5.0-3101-g310ab927149d  ***
\1b[8D\1b[J[00:00:00.423,370] \1b[0m<dbg> ethdrv.enc424j600_init_filters: ERXFCON: 0x005b\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.424,987] \1b[0m<dbg> ethdrv.enc424j600_init_phy: PHANA: 0x05e1\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.425,994] \1b[0m<dbg> ethdrv.enc424j600_init_phy: PHCON1: 0x1200\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.427,001] \1b[0m<dbg> ethdrv.enc424j600_init: EIE: 0x8850\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.427,398] \1b[0m<dbg> ethdrv.enc424j600_init: ECON1: 0x0001\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.427,825] \1b[0m<inf> ethdrv: ENC424J600 Initialized\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.428,192] \1b[0m<dbg> ethdrv.enc424j600_rx_thread: ESTAT: 0xda00\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.428,375] \1b[0m<inf> ethdrv: Link down\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.429,077] \1b[0m<dbg> ethdrv.enc424j600_rx_thread: ESTAT: 0x5a00\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.429,077] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0700\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.433,898] \1b[0m<inf> net_config: Initializing network\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.433,929] \1b[0m<inf> net_config: Waiting interface 1 (0x20000954) to be up...\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:11.434,326] \1b[1;31m<err> net_config: Timeout while waiting network interface\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:11.434,326] \1b[1;31m<err> net_config: Network initialization failed (-62)\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:11.434,478] \1b[0m<inf> net_dumb_http_srv_mt_sample: Waiting network to be connected\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m
```

I then tested with an nRF52840-DK with the same external olimax enc624j600 board, it seemed to work at startup but could not be reached over the network, so I unplugged the network cable and plugged it back in which it detected then came back with the an unknown interrupt error:
```
\00*** Booting Zephyr OS build zephyr-v2.5.0-3101-g310ab927149d  ***


\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.254,760] \1b[0m<dbg> ethdrv.enc424j600_init_filters: ERXFCON: 0x005b\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.255,187] \1b[0m<dbg> ethdrv.enc424j600_init_phy: PHANA: 0x05e1\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.255,432] \1b[0m<dbg> ethdrv.enc424j600_init_phy: PHCON1: 0x1200\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.255,706] \1b[0m<dbg> ethdrv.enc424j600_init: EIE: 0x8850\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.255,798] \1b[0m<dbg> ethdrv.enc424j600_init: ECON1: 0x0001\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.255,981] \1b[0m<dbg> ethdrv.enc424j600_rx_thread: ESTAT: 0xda00\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.256,042] \1b[0m<inf> ethdrv: Link down\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.256,134] \1b[0m<inf> ethdrv: ENC424J600 Initialized\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.262,115] \1b[0m<inf> net_config: Initializing network\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.262,115] \1b[0m<inf> net_config: Waiting interface 1 (0x20001470) to be up...\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.816,650] \1b[0m<dbg> ethdrv.enc424j600_rx_thread: ESTAT: 0xdf00\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.816,711] \1b[0m<inf> ethdrv: Link up\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.816,955] \1b[0m<dbg> ethdrv.enc424j600_setup_mac: PHANLPA: 0x85e1\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.817,230] \1b[0m<inf> ethdrv: 100Mbps\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.817,230] \1b[0m<inf> ethdrv: full duplex\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.817,413] \1b[0m<dbg> ethdrv.enc424j600_setup_mac: MACON2: 0x40b3\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.817,474] \1b[0m<dbg> ethdrv.enc424j600_setup_mac: MAMXFL (maximum frame length): 1518\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.817,535] \1b[0m<inf> ethdrv: Not suspended\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.817,810] \1b[0m<dbg> ethdrv.enc424j600_tx: pkt 0x2000f650 (len 106)\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.819,519] \1b[0m<dbg> ethdrv.enc424j600_tx: ETXSTAT: 0x0040\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.819,824] \1b[0m<dbg> ethdrv.enc424j600_tx: pkt 0x2000f650 (len 106)\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.821,533] \1b[0m<dbg> ethdrv.enc424j600_tx: ETXSTAT: 0x0040\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.821,746] \1b[0m<dbg> ethdrv.enc424j600_tx: pkt 0x2000f650 (len 78)\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.823,394] \1b[0m<dbg> ethdrv.enc424j600_tx: ETXSTAT: 0x0040\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.823,608] \1b[0m<dbg> ethdrv.enc424j600_tx: pkt 0x2000f650 (len 62)\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.825,195] \1b[0m<dbg> ethdrv.enc424j600_tx: ETXSTAT: 0x0040\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.825,286] \1b[0m<inf> net_config: Interface 1 (0x20001470) coming up\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.825,378] \1b[0m<inf> net_config: IPv4 address: 192.168.1.55\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.825,622] \1b[0m<inf> net_dumb_http_srv_mt_sample: Network connected\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.825,866] \1b[0m<dbg> net_dumb_http_srv_mt_sample.process_tcp6: Waiting for IPv6 HTTP connections on port 8080, sock 0\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.826,110] \1b[0m<dbg> net_dumb_http_srv_mt_sample.process_tcp4: Waiting for IPv4 HTTP connections on port 8080, sock 2\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:02.825,439] \1b[0m<dbg> ethdrv.enc424j600_tx: pkt 0x2000f5b0 (len 70)\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:02.827,056] \1b[0m<dbg> ethdrv.enc424j600_tx: ETXSTAT: 0x0040\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:03.827,301] \1b[0m<dbg> ethdrv.enc424j600_tx: pkt 0x2000f5b0 (len 70)\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:03.828,979] \1b[0m<dbg> ethdrv.enc424j600_tx: ETXSTAT: 0x0040\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m

\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:39.262,634] \1b[0m<dbg> ethdrv.enc424j600_rx_thread: ESTAT: 0xda00\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:39.262,695] \1b[0m<inf> ethdrv: Link down\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:39.262,847] \1b[0m<inf> net_dumb_http_srv_mt_sample: Network disconnected\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:40.892,669] \1b[0m<dbg> ethdrv.enc424j600_rx_thread: ESTAT: 0x5a00\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:40.892,700] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0708\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m
```

So out of 5 tests on 3 completely different boards, I'm not able to get a single one working with your baseline code.

lairdjm commented:
Not faring much better on a STM32 Nucleo L552ZE board:
```
*** Booting Zephyr OS build v2.6.0-rc1-61-gaadb337a7e20  ***


\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.004,000] \1b[0m<inf> ethdrv: Link down\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.004,000] \1b[0m<inf> ethdrv: ENC424J600 Initialized\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.009,000] \1b[0m<inf> net_config: Initializing network\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:00.009,000] \1b[0m<inf> net_config: Waiting interface 1 (0x20000a48) to be up...\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.558,000] \1b[0m<inf> ethdrv: Link up\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.558,000] \1b[0m<inf> ethdrv: 100Mbps\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.558,000] \1b[0m<inf> ethdrv: full duplex\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.558,000] \1b[0m<inf> ethdrv: Not suspended\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.558,000] \1b[0m<inf> net_config: Interface 1 (0x20000a48) coming up\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.558,000] \1b[0m<inf> net_config: IPv4 address: 192.168.1.55\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.558,000] \1b[0m<inf> net_dumb_http_srv_mt_sample: Network connected\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:01.559,000] \1b[0m<dbg> net_dumb_http_srv_mt_sample.process_tcp4: Waiting for IPv4 HTTP connections on port 8080, sock 0\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:15.023,000] \1b[1;31m<err> ethdrv: Maximum frame length exceeded\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:15.532,000] \1b[1;31m<err> ethdrv: Maximum frame length exceeded\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:16.533,000] \1b[1;31m<err> ethdrv: Maximum frame length exceeded\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:17.538,000] \1b[1;31m<err> ethdrv: Maximum frame length exceeded\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:18.517,000] \1b[1;31m<err> ethdrv: Maximum frame length exceeded\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:19.524,000] \1b[1;31m<err> ethdrv: Maximum frame length exceeded\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;31m--- 200 messages dropped ---
\1b[m[00:00:30.200,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.200,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.200,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.201,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.201,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.201,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.201,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.201,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.201,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.201,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.201,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.202,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.202,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.202,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.202,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.202,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.202,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.202,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.202,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.202,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.203,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.203,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.203,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.203,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.203,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.203,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.203,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.203,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.204,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.204,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.204,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.204,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;31m--- 274 messages dropped ---
\1b[m[00:00:30.668,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.668,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.668,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.668,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.668,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.668,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.669,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.669,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;31m--- 182 messages dropped ---
\1b[m[00:00:30.669,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.669,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.669,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.669,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.669,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.669,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.669,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.670,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.670,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.670,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.758,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.758,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.758,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.758,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;31m--- 193 messages dropped ---
\1b[m[00:00:30.758,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.758,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.758,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.759,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.759,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.759,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.759,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.759,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.759,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.759,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.905,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.905,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.905,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.905,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.905,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.905,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.906,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.906,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.906,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.906,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.906,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.906,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.906,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.906,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.907,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.907,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.907,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.907,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.907,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:30.907,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0600\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;31m--- 157 messages dropped ---
\1b[m[00:00:43.108,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;31m--- 1 messages dropped ---
\1b[m[00:00:43.108,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.108,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.108,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.108,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.109,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.109,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.109,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.109,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.109,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.109,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.109,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.110,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.110,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.110,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.110,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.110,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.110,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.110,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.110,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.110,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.111,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.111,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.111,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.111,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.111,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.111,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.111,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.111,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.112,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.112,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J[00:00:43.113,000] \1b[1;31m<err> ethdrv: Unknown Interrupt, EIR: 0x0000\1b[0m
\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m\1b[8D\1b[J\1b[1;32muart:~$ \1b[m
```

lairdjm commented:
@jukkar Can you review this fix please? I've tested it with an enc624j600 on nRF52840-dk and on nRF5340-dk, and with an enc424j600 on BL5340-DVK - boards are all visible on the network, can be pinged, can be accessed via the IP on port 8080 (using dumb_http_server_mt sample app), detect cable is present at startup and correctly detect cable has been inserted and removed.

jukkar commented:
> Can you review this fix please?

Probably my review is not very useful here as I am not familiar with this controller. Johan is the original author of this driver and you need to agree on what is the proper fix.

jfischer-no commented:
> I have an old olimax enc624j600 board which I attached to our BL5340 development board to use instead of the on-board enc424j600 to exclude any possible issues with the components on our board, this made no difference and I get the unknown interrupt error whilst at the same time someone else remove all the intermediatory components on our development board between the BL5340 and enc424j600 e.g. for level shifting which also made no difference. I then tested with an nRF5340-DK and the olimax enc624j600 and I see the exact same issue with the original code - it does not work:

I tested today: link board ETH with nRF52840DK, link board ETH with frdm_k64f, mikroETH100 with nRF52840. I can not reproduce issue that you describe. On BL5340, can you please connect AD0..AD14 and PSPCFG0 pins to GND and retest?

lairdjm commented:
We have tested changing our interrupt pull-up resistor from 100K to 10K as per your board and grounded all AD pins (PSP was already grounded) and it has made no difference.

jukkar commented:
Are there different versions of enc424j600 which could explain the difference you are seeing?

lairdjm commented:
> Are there different versions of enc424j600 which could explain the difference you are seeing?

Microchip's errata document seems to indicate only 1 silicon revision: A2

lairdjm commented:
After further debugging it seems that the unknown interrupt was caused by the interrupt being triggered prior to the driver configuring the interrupt sources it wanted to receive, disabling interrupts until they have been configured has resolved the issue. Additionally given that no other network driver in zephyr nor the encx24j600 driver in the linux kernel operates in the way that this driver did operate in and given that it doesn't make any sense for a production IoT firmware (whereby if an unknown interrupt was received then this driver would essentially kill itself with no warnings by not re-enabling interrupts) I have moved the option to do that to be behind a Kconfig option which is opt-in

jfischer-no commented on drivers/ethernet/eth_enc424j600.c:
```diff
@@ -450,8 +451,7 @@ static void enc424j600_rx_thread(struct enc424j600_runtime *context)
 
-		enc424j600_clear_sfru(context->dev, ENC424J600_SFR3_EIEL,
-				      ENC424J600_EIE_INTIE);
+		enc424j600_write_sbc(context->dev, ENC424J600_1BC_CLREIE);
```
How this change is relevant to the bug?

jfischer-no commented on drivers/ethernet/eth_enc424j600.c:
```diff
@@ -484,11 +484,13 @@ static void enc424j600_rx_thread(struct enc424j600_runtime *context)
 
-		enc424j600_set_sfru(context->dev, ENC424J600_SFR3_EIEL,
-				    ENC424J600_EIE_INTIE);
+		enc424j600_write_sbc(context->dev, ENC424J600_1BC_SETEIE);
```
How this change is relevant to the bug?

jfischer-no commented on drivers/ethernet/eth_enc424j600.c:
```diff
@@ -684,6 +686,8 @@ static int enc424j600_init(const struct device *dev)
 		return -EIO;
 	}
 
+	enc424j600_write_sbc(context->dev, ENC424J600_1BC_CLREIE);
```
Actually it is simpler to change 
```
enc424j600_set_sfru(dev, ENC424J600_SFR3_EIEL,
      			    ENC424J600_EIE_PKTIE | ENC424J600_EIE_LINKIE);
```

to 
```
enc424j600_write_sfru(dev, ENC424J600_SFR3_EIEL,
      			    ENC424J600_EIE_PKTIE | ENC424J600_EIE_LINKIE);
```

and place it in this line. This is what I wrote in the email as an example, all this is a bit too exhausting, I will open an alternative PR.

galak commented:
Closing this as we merged a different PR solution to the issue.
